### PR TITLE
Update design of template + new SDK version

### PR DESCRIPTION
**Pull request by jacc**
This PR includes the following changes
- Bump SDK version to 0.0.4, the latest available
- Redesign the template to use Vercel's newest Next.js 13 design
- fixes INFRA-60

### CODE REVIEW

File: app/app/ssg/product-gated/page.tsx [hunk 1]
```diff
@@ -1,12 +1,58 @@
...
 const Page: NextAppPage = () => {
-  return <>Product owned!</>;
+  return (
+    <>
+      <div className={styles.container}>
+        <Head>
+          <title>Paywall App</title>
+          <meta name="description" content="Generated by create next app" />
+          <link rel="icon" href="/logo.svg" />
+        </Head>
```
Comment (baked-dev) [inline app/app/ssg/product-gated/page.tsx]: the next.js app directory uses a special metadata export for setting these tags.
see: https://beta.nextjs.org/docs/api-reference/metadata

File: app/app/ssg/product-gated/page.tsx [hunk 2]
```diff
@@ -1,12 +1,58 @@
+"use client";
```
Comment (baked-dev) [inline app/app/ssg/product-gated/page.tsx]: does this not work as a server component?

File: app/app/ssr/product-gated/PurchaseLink.tsx [hunk 3]
```diff
@@ -3,6 +3,10 @@
 import { usePurchaseLink } from "@/lib/get-purchase-link";
 import Link from "next/link";
 import { FunctionComponent, PropsWithChildren } from "react";
+import styles from "../../../../styles/Home.module.css";
+import { Inter } from "next/font/google";
+
+const inter = Inter({ subsets: ["latin"] });
```
Comment (baked-dev) [inline app/app/ssr/product-gated/PurchaseLink.tsx]: this appears to be unused

**Review by baked-dev — CHANGES_REQUESTED** [review]

File: app/app/ssg/product-gated/page.tsx (hunk 1 above)
Comment (baked-dev) [reply app/app/ssg/product-gated/page.tsx]: you might have to update the version of next we use to enable this feature, its been introduced in 13.2

**Comment by stevo2105** [comment]
#### 📝 Changed routes:


| [**/**](https://next-template-1vquqzkxs-whop.vercel.app/) | [**/ssg/product-gated**](https://next-template-1vquqzkxs-whop.vercel.app/ssg/product-gated) |
|:-:|:-:|
|<a href="https://next-template-1vquqzkxs-whop.vercel.app/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/NJ-GcHwSp9c5ytT0C2QMHiKmYY815xpiQdADlXkQGUf6DleEN-RNO-z02yuv1SqrhittK86qKxR5QpmU833ajw.png">(view full size)</a> |<a href="https://next-template-1vquqzkxs-whop.vercel.app/ssg/product-gated">Screenshot of /ssg/product-gated</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/_cr7pr2NIad1bchctsYcXD4JziCq0xvHBHFGWDnkS8X7o2x8VccCJFl-Zu-EVRwn_3859vbiICf_FDd7RvgwOw.png">(view full size)</a> |



| [**/ssr**](https://next-template-1vquqzkxs-whop.vercel.app/ssr) | [**/ssr/logged-in**](https://next-template-1vquqzkxs-whop.vercel.app/ssr/logged-in) |
|:-:|:-:|
|<a href="https://next-template-1vquqzkxs-whop.vercel.app/ssr">Screenshot of /ssr</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/WZK-bbqbT6sZ_cyh8csSk1RxkWYINT7j3UrcUWKIKXEMCUAEtcVjsHCot_DvoUdZOjOEfaUK5qZf5wSpySc3IA.png">(view full size)</a> |<a href="https://next-template-1vquqzkxs-whop.vercel.app/ssr/logged-in">Screenshot of /ssr/logged-in</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/QoM2pxsUrvzLueB1eTSShD0gJ2SHR6gjr1HOZ0lV0n3xixvqQyBonG3bXjhxQZVKzt5ugc77ZojszpfMh6EFUg.png">(view full size)</a> |



| [**/ssr/product-gated**](https://next-template-1vquqzkxs-whop.vercel.app/ssr/product-gated) |
|:-:|
|<a href="https://next-template-1vquqzkxs-whop.vercel.app/ssr/product-gated">Screenshot of /ssr/product-gated</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/xzefzziYTfOdIzoYHGcanOKLEYdi9clcpOtLmrz3FDAZ-QzYJMdQ6evL0WtwwRLsrfv81VWqk-VIiU6cxk0IQg.png">(view full size)</a> |







Commit a641154e719f94db2bf488bb651bc188464b3a36 (https://next-template-1vquqzkxs-whop.vercel.app).

File: app/app/ssg/product-gated/page.tsx (hunk 2 above)
Comment (jacc) [reply app/app/ssg/product-gated/page.tsx]: it didn't before, but now removing it it does 🥴

File: app/app/ssr/product-gated/layout.tsx [hunk 4]
```diff
@@ -35,56 +38,78 @@ export default async function SSRProductGatedLayout({
...
-            <p style={{ textAlign: "center" }}>User has membership: No</p>
-          </main>
-
-          <footer className={styles.footer}>
+        <Head>
+          <title>Whop powered application</title>
+          <meta name="description" content="Generated by create next app" />
+          <meta name="viewport" content="width=device-width, initial-scale=1" />
+          <link rel="icon" href="/logo.svg" />
+        </Head>
```
Comment (baked-dev) [inline app/app/ssr/product-gated/layout.tsx]: please move this to metadata (or a head.tsx file if you dont want to update the next version yet)

File: app/app/ssr/product-gated/page.tsx [hunk 5]
```diff
@@ -27,52 +30,63 @@ const Page: NextAppPage = async () => {
         <Head>
           <title>Paywall App</title>
           <meta name="description" content="Generated by create next app" />
-          <link rel="icon" href="/favicon.ico" />
+          <link rel="icon" href="/logo.svg" />
         </Head>
```
Comment (baked-dev) [inline app/app/ssr/product-gated/page.tsx]: this is redundant as its already been set in the layout.tsx in the same segment

**Review by baked-dev — CHANGES_REQUESTED** [review]

File: app/app/ssr/product-gated/layout.tsx (hunk 4 above)
Comment (baked-dev) [reply app/app/ssr/product-gated/layout.tsx]: just saw that its been updated to 12.2.4 so it should go into metadata

**Review by baked-dev — APPROVED** [review]

**Review by baked-dev — APPROVED** [review]

**Pull request merged**